service/backend/pchain/block.go

```diff
@@ -30,97 +31,114 @@ func (b *Backend) Block(ctx context.Context, request *types.BlockRequest) (*type
 		hash = *request.BlockIdentifier.Hash
 	}
 
-	isGenesisBlockRequest := b.isGenesisBlockRequest(blockIndex, hash)
-	if isGenesisBlockRequest {
+	var (
+		blkIdentifier       *types.BlockIdentifier
+		parentBlkIdentifier *types.BlockIdentifier
+		blkTime             int64
+		rTxs                []*types.Transaction
+		metadata            map[string]interface{}
+	)
+
+	if b.isGenesisBlockRequest(blockIndex, hash) {
 		genesisTxs, err := b.getFullGenesisTxs()
 		if err != nil {
 			return nil, service.WrapError(service.ErrClientError, err)
 		}
-		rosettaTxs, err := b.parseRosettaTxs(genesisTxs, nil)
+		rosettaTxs, err := parseRosettaTxs(b.txParserCfg, blocks.GenesisCodec, genesisTxs, nil)
```

Comment:
We should always use `blocks.GenesisCodec` instead of `blocks.Codec` to parse genesis and its transactions. Reason is we allow genesis content to be larger than the maximum size an ordinary block/transaction can have

Reply:
Yeah, this got me before since testnet genesis is small enough to fit into the `blocks.Codec` and we only found out about this after trying to parse mainnet blocks 😄